### PR TITLE
Putting an end to the endless cryptonerd comments by recommending people to use the Mozilla SSL Configuration Generator

### PR DESCRIPTION
**Pull request by Aldarone**
We're not security expert and we shouldn't have to be. So it's better to tell people where to go to find some help.

### CODE REVIEW

**Comment by Technowix** [comment]
Would like to add that the "Fucking Cryptonerd" configuration actually affect 7.2% of the Androids phone running according to 
https://developer.android.com/about/dashboards/index.html#Platform

**Comment by Angristan** [comment]
> (Since >kitkat support TLS1.2)

That's the thing. Backward support isn't about the cipher here, it's about the TLS version.

Put `EECDH+AESGCM:EECDH+CHACHA20:EECDH+AES` on https://cryptcheck.fr/suite and you will see the cipher is compatible with Android 4.0, IE7, iOS 5, Windows XP... While still being safe. I can't see how Mastodon would work in anything older than that...

There is no hint AES will be broken soon and if it is we're fucked in any case. As for CHACHA20, it' still new, and if you use it on an OpenSSL 1.0 system, which does not support it, it's simply ignored without causing any error.

Drawback of this cipher: none. It's not like a new algorithm will be added tomorrow, and even if it, would have to wait years before seeing it on Debian.

So https://github.com/tootsuite/documentation/commit/72ed10378f4c93eecc901f7e947d7f0508f6345a is useless in termes of useful compatibility and it adds nothing but flaws to the HTTPS config (DSS, DH...).

Concerning [`ssl_ecdh_curve`](https://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_ecdh_curve), we should not specify it (as I did before https://github.com/tootsuite/documentation/commit/72ed10378f4c93eecc901f7e947d7f0508f6345a above, so I approve this change), but probably not for the reason you think.

Before the commit, we were using P-256, which does not have any compat issue (even though the author says it has with Android 7, that's not true which makes it look like he really tested his config...).
Why we shouldn't specify one ? Before Nginx 1.11, the default one was `P-256`, so no problem, and since, the default value is `auto`, which equals to  `X25519:P-256:P-521:P-384` with OpenSSL 1.1 and `P-256:P-521:P-384` with older versions. So no compatibility nor security problems here, nothing we nor the users have to worry about. 

And concerning TLS, we have 2 possibility for the doc : 

```
# Add TLSv1.1 and TLSv1.0 for Android 4.0 compatibility 
ssl_protocols TLSv1.2; 
```

or 

```
# Remove TLSv1.0 and TLSv1.1 if you don't need Android 4.0 compatibility
ssl_protocols TLSv1.2 TLSv1.1 TLSv1.0;
```

TL;DR : Secure, simple, compatible and pretty future-proof config : 

```
# Add TLSv1.1 and TLSv1.0 for Android 4.0 compatibility 
ssl_protocols TLSv1.2; 
ssl_ciphers EECDH+AESGCM:EECDH+CHACHA20:EECDH+AES;
```

[How it looks on SSL Labs](https://www.ssllabs.com/ssltest/analyze.html?d=mstdn.io&s=164.132.78.232) (see the compatibility section)

And that's all. Compatible with every version of Nginx, and OpenSSL 1.x, so bascially everything. And if the admin wants to use Android 4 phones, he just had to read one line. I can't see what would be better than this ?



I'm not playing the cryptonerd here, this would be better and easier for everyone, and should not require any modification for years.

What do you think?

*btw, of course I'm not an expert on the subject but there are no personal preferences in this message, just facts, so I'm pretty confident of what I said. Even so, don't hesitate to correct me.*

**Comment by lanodan** [comment]
I don’t agree with Alda:
* There should be sane defaults, there is no use to break things by default, same goes for crypto’, with the limit that we should not have broken ciphers (RC4, 3DES, …)
* We should not link to something like Mozilla as they’re not neutral on this. (And Mozilla privacy on Firefox is a lie, why should we blindly trust them…)
